File: Back-end/BookStore.Domain/Entities/Order.cs
using BookStore.Domain.Enums;
using System;
using System.Collections.Generic;

namespace BookStore.Domain.Entities;
public class Order : IEntity
{
    public int Id { get; set; }
    public OrderState State { get; set; } = OrderState.Placed;
    public string UserName { get; set; }
    public string Email { get; set; }
    public string PhoneNumber { get; set; }
    public string Address { get; set; }
    public OrderReceiptMethod OrderReceiptMethod { get; set; }
    public PaymentMethod PaymentMethod { get; set; }
    public DateTime PlacedDate { get; set; }
    public DateTime? DeliveredDate { get; set; }
    
    public User User { get; set; }
    public int UserId { get; set; }

    public IEnumerable<OrderProduct> Products { get; set; } = new List<OrderProduct>();
}

File: Front-end/store/src/app/core/models/order.ts
import { BasketProduct } from './basket-product';

export class Order {
  public id: number;

  public state: string;

  public userName: string;

  public email: string;

  public phoneNumber: string;

  public address: string;

  public orderReceiptMethod: string;

  public paymentMethod: string;

  public placedDate: string;

  public deliveredDate: string;

  public products: BasketProduct[];


  constructor(order: Order) {
    this.id = order.id;
    this.state = order.state;
    this.userName = order.userName;
    this.email = order.email;
    this.phoneNumber = order.phoneNumber;
    this.address = order.address;
    this.orderReceiptMethod = order.orderReceiptMethod;
    this.paymentMethod = order.paymentMethod;
    this.placedDate = order.placedDate;
    this.deliveredDate = order.deliveredDate;
    this.products = order.products;
  }
}
